Just synchronizing with my local fixes before moving the debugger window to a separate static library

dartfnm opened the pull request.

dartfnm committed:
Introduces a very early debugger feature for quaesar

+ EASTL lib
+ added some not very functional debugger windows
+ CMake flags for VC Windows development

dartfnm committed:
removed old cpp files & format fixed

make clang & gcc happy

gcc compile errors fixed

dartfnm committed:
Fix ImGui cmake file for SDL include path

dartfnm committed:
Merge branch 'debugger-start' of https://github.com/DartFNM/quaesar into debugger-start

dartfnm committed:
Debugger v0.2

    Debugger v0.2

    - Windows: Disasm, copper, blitter, custom regs, scr_graph, console;
    - Shortcuts, Actions, Log, Console, messages;
    - Debug trace: Step-In, Step Copper, Skip N scanlines;
    - Separate thread for UAE
    - Update imgui docking to latest ver 1.91.3
    - and many, many ..

Fix creation of UAE window and process events in main thread

- Due to bug on MacOS: NSWindow should only be instantiated on the main thread!
- A known bug is that shortcuts are now global instead of separate for each window.

dartfnm committed:
Fixed most of the memory leak issues

Compatible with old compilers since the minimum C++11 version

dartfnm committed:
Fix bug with UAE semaphore's SDL implementation

dartfnm committed:
Fix UAE non-deallocated memory cleanup on quit

dartfnm committed:
Merge branch 'theblacklotus:debugger-start' into debugger-start

dartfnm committed:
Command lines same as in WinUAE

dartfnm committed:
refactor: Cleanup and rename some vars

dartfnm committed:
refactor: Rename some vars

dartfnm committed:
refactor: Rename some vars

dartfnm committed:
Merge branch 'theblacklotus:debugger-start' into debugger-start

dartfnm committed:
Merge branch 'debugger-start' of https://github.com/DartFNM/quaesar into debugger-start

dartfnm committed:
Merge branch 'debugger-start' of https://github.com/DartFNM/quaesar into debugger-start

emoon commented:
Sorry for the delay. I have run the tests now and there seems to be some issues with the change.

dartfnm committed:
Fixes for cross platform build

- `file_system.cpp` should compile on all platforms, to support `-s filesystem=rw,dh0:c:\\Amiga\\hd0` command line;
- Added cmake's target for `clang-format` to pass the `Clang Forma Check` action;
- Moved the defines from CMakeLists.txt to the `sysconfig.h`

dartfnm committed:
Merge branch 'debugger-start' of https://github.com/DartFNM/quaesar into debugger-start

dartfnm commented:
I fixed compilation warnings with __FUNCTION__ macro for clang, and formatting errors.
But I seem to have completely forgotten that I included the original UAE implementation in src/file_system.cpp, so that in quasar we can also mount the local folder as an hd0 partition. I'll have to port it to SDL more carefully sometime.

emoon commented:
Thanks! Seems there are still some issues building the code tho

dartfnm committed:
Merge branch 'debugger-start' of https://github.com/DartFNM/quaesar into debugger-start

emoon commented:
Thanks for the fixes!
